elastic/kibana
# [Security Solution][Detection Engine] Only check read privileges of extant indices during rule execution

## Summary

This PR modifies the privilege-checking behavior during rule execution, restricting the indices against which we verify `read` access to only those that exist.

### Outstanding questions

- [x] Are there any backwards-compatibility/semver concerns with changing this behavior?
    * We discussed in which situations a user might reasonably be using the existing behavior, and determined those to be borderline. If we end up receiving feedback to the contrary, we can add back the old behavior as configuration.
- [x] Is the `IndexPatternsFetcher` an appropriate implementation to use for the existence checking?


### Steps to Review
1. Create a rule with a pattern including a non-existent index, e.g. `auditbeat-*,does-not-exist` 
2. Enable the rule, and observe no warning about e.g. missing read privileges for `does-not-exist`
3. (optional) Remove read access to `auditbeat-*`, or extend the pattern to include an existing index that the rule author cannot read
4. (optional) Observe a warning for the non-readable index


### Checklist

Delete any items that are not applicable to this PR.


### For maintainers
